static/modes/cobol-mode.ts

```diff
@@ -0,0 +1,1183 @@
+// Copyright (c) 2023, Compiler Explorer Authors
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+import * as monaco from 'monaco-editor';
+
+function definition(): monaco.languages.IMonarchLanguage {
+    return {
+        ignoreCase: true,
+        keywords: [
+            '3-D',
```

Comment:
Wow that's a lot of keywords!

Reply:
That's what you get from `cobc --list-reserved --list-intrinsics` ;-)

The issue you always have with COBOL "in the wild" are the dialects. You have the many keywords from the standard,then a lot of additional ones from COBOL implementations - most of the additional ones in the list here are from ACUCOBOL-GT.

Often many of the reserved words (at least with GnuCOBOL) will be context-sensitive, so for code highlighting one _may_ consider to limit those to the standard ones by `cobc `-std=cobol2014 --list-reserved --list-intrinsics`.

... thinking of this: it likely would be ideal to have the dialects as part of the options to choose which then also adjust the syntax highlighting.

Note that different dialects also will yield different results, starting from "not compile" ending in "different result at runtime". It is _similar_ to `-std` in gcc, just _much more extreme_.  But I did not found an option on godbold to toggle `-std` easily for GCC either...